kernels/simple_mult/main.c

```diff
@@ -0,0 +1,41 @@
+#include "data.h"
+#include "stdint.h"
+
+#include <snrt.h>
+
+// Kernel provided via external definition
+void simple_mult(int32_t *a, int32_t *b, int32_t *d);
+
+int main() {
+    // Allocate shared local memory
+    // By avoiding allocators and bumping by a known offset a base pointer
+    // (snrt_l1_next()) that is the same for all the cores in the cluster, we are
+    // essentially providing the same memory regions to all the cores in this cluster.
+    int32_t *local_A = (int32_t*)snrt_l1_next();
+    int32_t *local_B = local_A + N;
+    int32_t *local_D = local_B + N;
+
+    // Copy data in shared local memory
+    if (snrt_is_dm_core()) {
+        snrt_dma_start_1d(local_A, A, N * sizeof(float));
+        snrt_dma_start_1d(local_B, B, N * sizeof(float));
+    }
+
+    snrt_cluster_hw_barrier();
+
+    // Launch kernel: from this point on only core 0 is required to be alive.
+    int thiscore = snrt_cluster_core_idx();
+    if (thiscore != 0) return 0;
```

Comment:
Yes, but this is for running on the default snitch_cluster, and in this example we only use core 0.
I agree that this is not the most robust thing; you can change the hjson to be anything and there's no way this code will check for that.
That raises an important question for you though @rgantonio : Are core id's in the snitch template assigned by the order in which they appear in the hjson fil?

Reply:
@JosseVanDelm yes they are assigned in order. The top is 0 and increases index number by 1 as you go down.

It's funny because the `snrt_is_compute_core()` is true if the core ID is less than the number of cores. So say if you have 8 compute cores but the core IDs are index from 0-7, then if you have a core ID 3, and it's < 8, then you are a compute core. Xiaolin and I just found out about this 😄 

Core ID 8, (the 9th core) becomes the DMA.

Bottomline, the order in hjson file indicates the core ID, starting from 0 from the top and as you go down you increment core ID.

Reply:
Yes the routines in the runtime that check this are not very robust, would be better if they would read it directly from some register instead :D